lib/activerecord-import/import.rb

```diff
@@ -367,7 +367,7 @@ def import_helper( *args )
           column_names = self.column_names.dup
         end
 
-        if column_names.include?(primary_key) && columns_hash[primary_key].type == :uuid
+        if models.first.id.nil? && column_names.include?(primary_key) && columns_hash[primary_key].type == :uuid
```

Comment:
Hm maybe an error is a better result, since that should be an exceptional scenario if some IDs have values and some do not.